include/zephyr/dt-bindings/clock/stm32wb_clock.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2022 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32WB_CLOCK_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32WB_CLOCK_H_
+
+/** Bus clocks */
+#define STM32_CLOCK_BUS_AHB1    0x048
+#define STM32_CLOCK_BUS_AHB2    0x04c
+#define STM32_CLOCK_BUS_AHB3    0x050
+#define STM32_CLOCK_BUS_APB1    0x058
+#define STM32_CLOCK_BUS_APB1_2  0x05c
+#define STM32_CLOCK_BUS_APB2    0x060
+
+#define STM32_PERIPH_BUS_MIN	STM32_CLOCK_BUS_AHB1
+#define STM32_PERIPH_BUS_MAX	STM32_CLOCK_BUS_APB2
+
+/** Peripheral clock sources */
+/* RM0434, § Clock configuration register (RCC_CCIPRx) */
+
+/** Fixed clocks  */
+#define STM32_SRC_HSI		0x001
+/* #define STM32_SRC_HSI48	0x002 */
+#define STM32_SRC_LSE		0x003
+#define STM32_SRC_LSI		0x004
+#define STM32_SRC_MSI		0x005
+/** System clock */
+#define STM32_SRC_SYSCLK	0x006
+/** Bus clock */
+#define STM32_SRC_PCLK		0x007
+/** PLL clocks */
+#define STM32_SRC_PLLCLK	0x008
+/* TODO: PLLSAI clocks */
+
+#define STM32_SRC_CLOCK_MIN	STM32_SRC_HSI
+#define STM32_SRC_CLOCK_MAX	STM32_SRC_PLLCLK
+
+/**
+ * @brief STM32 clock configuration bit field.
+ *
+ * - reg   (1/2/3)         [ 0 : 7 ]
+ * - shift (0..31)         [ 8 : 12 ]
+ * - mask  (0x1, 0x3, 0x7) [ 13 : 15 ]
+ * - val   (0..7)          [ 16 : 18 ]
+ *
+ * @param reg RCC_CCIPRx register offset
+ * @param shift Position within RCC_CCIPRx.
+ * @param mask Mask for the RCC_CCIPRx field.
+ * @param val Clock value (0, 1, ... 7).
+ */
+
+#define STM32_CLOCK_REG_MASK    0xFFU
+#define STM32_CLOCK_REG_SHIFT   0U
+#define STM32_CLOCK_SHIFT_MASK  0x1FU
+#define STM32_CLOCK_SHIFT_SHIFT 8U
+#define STM32_CLOCK_MASK_MASK   0x7U
+#define STM32_CLOCK_MASK_SHIFT  13U
+#define STM32_CLOCK_VAL_MASK    0x7U
+#define STM32_CLOCK_VAL_SHIFT   16U
+
+#define STM32_CLOCK(val, mask, shift, reg)					\
+	((((reg) & STM32_CLOCK_REG_MASK) << STM32_CLOCK_REG_SHIFT) |		\
+	 (((shift) & STM32_CLOCK_SHIFT_MASK) << STM32_CLOCK_SHIFT_SHIFT) |	\
+	 (((mask) & STM32_CLOCK_MASK_MASK) << STM32_CLOCK_MASK_SHIFT) |		\
+	 (((val) & STM32_CLOCK_VAL_MASK) << STM32_CLOCK_VAL_SHIFT))
+
+/** @brief RCC_CSR register offset */
```

Comment:
Comment references `RCC_CSR` but related to `RCC_CCIPR`.

The `RCC_CSR` has `RFWKP_SEL` that might be added later on.